scene/3d/gpu_particles_collision_3d.cpp

```diff
@@ -347,7 +347,9 @@ void GPUParticlesCollisionSDF3D::_compute_sdf(ComputeSDFParams *params) {
 	WorkerThreadPool::GroupID group_task = WorkerThreadPool::get_singleton()->add_template_group_task(this, &GPUParticlesCollisionSDF3D::_compute_sdf_z, params, params->size.z);
 	while (!WorkerThreadPool::get_singleton()->is_group_task_completed(group_task)) {
 		OS::get_singleton()->delay_usec(10000);
-		bake_step_function(WorkerThreadPool::get_singleton()->get_group_processed_element_count(group_task) * 100 / params->size.z, "Baking SDF");
+		if (bake_step_function) {
+			bake_step_function(WorkerThreadPool::get_singleton()->get_group_processed_element_count(group_task) * 100 / params->size.z, "Baking SDF");
+		}
```

Comment:
Doesn't this lead to an infinite loop, waiting for the completion of a group task which is never started?

Reply:
I tried baking the SDF in editor after setting `GPUParticlesCollisionSDF3D::bake_step_function = nullptr;` in the `GPUParticlesCollisionSDF3DEditorPlugin` constructor. The editor progress bar doesn't update but the baking completes as normal.

The group task is started above the while loop. The loop only busy waits until the task finishes and provides progress updates, it doesn't progress any work.